### ISSUE TITLE
Installation on Apple M2

### ISSUE THREAD

**Issue by DilhanM** [block-hczhai/block2-preview#41]
I'm attempting to install block2 on an Apple M2 and running into a mkl dependency issue. Can you please help resolve it?

`ERROR: Cannot install block2==0.1.10, block2==0.1.4, block2==0.1.5, block2==0.1.6, block2==0.1.7, block2==0.1.8, block2==0.2.0, block2==0.3.0, block2==0.4.0, block2==0.4.1, block2==0.4.10, block2==0.4.12, block2==0.4.13, block2==0.4.14, block2==0.4.2, block2==0.4.5, block2==0.4.6, block2==0.4.8, block2==0.4.9, block2==0.5.0 and block2==0.5.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    block2 0.5.1 depends on mkl==2021.4
    block2 0.5.0 depends on mkl==2019
    block2 0.4.14 depends on mkl==2019
    block2 0.4.13 depends on mkl==2019
    block2 0.4.12 depends on mkl==2019
    block2 0.4.10 depends on mkl==2019
    block2 0.4.9 depends on mkl==2019
    block2 0.4.8 depends on mkl==2019
    block2 0.4.6 depends on mkl==2019
    block2 0.4.5 depends on mkl==2019
    block2 0.4.2 depends on mkl==2019
    block2 0.4.1 depends on mkl==2019
    block2 0.4.0 depends on mkl==2019
    block2 0.3.0 depends on mkl==2019
    block2 0.2.0 depends on mkl==2019
    block2 0.1.10 depends on mkl
    block2 0.1.8 depends on mkl
    block2 0.1.7 depends on mkl
    block2 0.1.6 depends on mkl
    block2 0.1.5 depends on mkl
    block2 0.1.4 depends on mkl`

I have the following packages installed on my conda environment.

Name                    Version                   Build  Channel
bzip2                     1.0.8                h1de35cc_0  
ca-certificates     2023.01.10           hecd8cb5_0  
intel-openmp       2021.4.0          hecd8cb5_3538  
libffi                    3.4.2                hecd8cb5_6  
mkl                       2021.4.0           hecd8cb5_637  
ncurses                   6.4                  hcec6c5f_0  
openssl                   1.1.1t               hca72f7f_0  
pip                       23.0.1          py310hecd8cb5_0  
python                    3.10.11              h218abb5_2  
readline                  8.2                  hca72f7f_0  
setuptools                66.0.0          py310hecd8cb5_0  
sqlite                    3.41.2               h6c40b1e_0  
tk                        8.6.12               h5d9f67b_0  
tzdata                    2023c                h04d1e81_0  
wheel                     0.38.4          py310hecd8cb5_0  
xz                        5.4.2                h6c40b1e_0  
zlib                      1.2.13               h4dc903c_0  

I am on an Apple Mac M2, on Ventura 13.3.1 (a).

**Comment by hczhai** [comment]
Thanks for your interest in using block2. To have a better understanding of the problem, may I ask you a few more details:
(1) Are you running in (a) the Rosetta (x86) mode on an Apple Mac M2; (b) the native arm64 mode?
(2) What is the command that you are using for installing block2? For example, it can be ``pip install block2`` or something else.
(3) Please let me know what you can get if you execute ``pip install mkl==2021.4`` before installing block2.

**Comment by DilhanM** [comment]
Thank you for the prompt reply. I've tried to install using both Rosetta and arm64 modes.

1. To try the rosetta (x86) mode, I created a conda environment using

`CONDA_SUBDIR=osx-64 conda create -n myenv_x86 python=3.7
conda activate myenv_x86
conda config --env --set subdir osx-64`

On this environment, I first tried to install block2 using `pip install block2`. This failed due to the error I reported above. Then, I tried `pip install mkl==2021.4` on this environment, but pip is unable to find this environment. It returns the error `ERROR: Could not find a version that satisfies the requirement mkl==2021.4 (from versions: none)`. However, I managed to install `mkl` using `conda install mkl==2021.4`. After this installation, `conda list` shows `mkl   2021.4.0   hecd8cb5_637`.
Then I proceeded to try install block2 using `pip install block2`, but I still encounter the above dependency error.

2. To try the native arm64 mode, I created a conda environment using

`CONDA_SUBDIR=osx-arm64 conda create -n myenv_arm64 -c conda-forge
conda activate myenv_arm64
conda config --env --set subdir osx-arm64`

On this conda environment, I cannot install mkl using either pip or conda. `conda install -c anaconda mkl` fails to find the package on any of the channels.

Thank you so much for your time.
